plugins/scaffolder-backend/src/scaffolder/tasks/types.ts

```diff
@@ -190,10 +190,34 @@ export interface TaskStore {
     tasks: { taskId: string }[];
   }>;
 
+  list?(options: {
+    filters?: {
+      createdBy?: string | string[];
+      status?: TaskStatus | TaskStatus[];
+    };
+    pagination?: {
+      limit?: number;
+      offset?: number;
+    };
+    order?: { order: 'asc' | 'desc'; field: string }[];
+  }): Promise<{ tasks: SerializedTask[]; totalTasks?: number }>;
+
+  /**
+   * @deprecated Make sure to pass `createdBy` and `status` in the `filters` parameter instead
+   */
   list?(options: {
     createdBy?: string;
     status?: TaskStatus;
-  }): Promise<{ tasks: SerializedTask[] }>;
+    filters?: {
```

Comment:
You don't have to add the additional fields in here, you can just trim this to the old options and the old API. This @deprecated one should be exactly the same as the old `list` method, and the new one above with the new options.